apps/coscrad-frontend/src/components/resources/spatial-features/spatial-feature-detail.full-view.presenter.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+    GeometricFeatureType,
+    ICategorizableDetailQueryResult,
+    ISpatialFeatureViewModel,
+} from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { Grid, styled } from '@mui/material';
+import { SinglePropertyPresenter } from '../../../utils/generic-components';
+import { ResourceNamePresenter } from '../../../utils/generic-components/presenters/resource-name-presenter';
+import { CoscradMainContentContainer } from '../../../utils/generic-components/style-components/coscrad-main-content-container';
+import { FunctionalComponent } from '../../../utils/types/functional-component';
+import { LineTextPresenter } from './thumbnail-presenters/line-text-presenter';
+import { PointTextPresenter } from './thumbnail-presenters/point-text-presenter';
+import { PolygonTextPresenter } from './thumbnail-presenters/polygon-text-presenter';
+
+interface HasCoordinates<T = unknown> {
+    coordinates: T;
+}
+
+const lookupTable: { [K in GeometricFeatureType]: FunctionalComponent<HasCoordinates> } = {
+    [GeometricFeatureType.point]: PointTextPresenter,
+    [GeometricFeatureType.line]: LineTextPresenter,
+    [GeometricFeatureType.polygon]: PolygonTextPresenter,
+};
+
+const StyledPlaceIcon = styled('img')({
+    width: '150px',
+});
+
+export const SpatialFeatureDetailFullViewPresenter = (
+    spatialFeature: ICategorizableDetailQueryResult<ISpatialFeatureViewModel>
+): JSX.Element => {
+    const { id, geometry, properties } = spatialFeature;
+
+    if (!geometry) {
+        throw new Error(`Spatial Feature: ${id} is missing geometry definition`);
+    }
+
+    if (!properties) {
+        throw new Error(`Spatial Feature: ${id} is missing its properties`);
+    }
+
+    const { name, description, imageUrl } = properties;
+
+    const { type: geometryType, coordinates } = geometry;
+
+    const CoordinatesTextPresenter = lookupTable[geometryType];
+
+    if (isNullOrUndefined(CoordinatesTextPresenter)) {
+        throw new Error(
+            `There is no coordinates text presenter registered for coordinates of geometry type: ${geometryType}`
+        );
+    }
+
+    return (
+        <CoscradMainContentContainer>
+            <Grid container spacing={0}>
+                <Grid item xs={3}>
+                    <div data-testid={id} />
+                    {/* Preview will eventually include images taken from video or photos, etc. */}
+                    <StyledPlaceIcon src={imageUrl} alt={`Spatial Feature ${id}`} />
```

Comment:
It's a bit awkward that you call the component an Icon but you pass in the imageURL which makes it really more of an image than an icon. I guess we can live with this though, because the image really is playing the role of an icon in the map.